prow/cmd/deck/main.go

```diff
@@ -226,43 +227,53 @@ func handleLog(lc logClient) http.HandlerFunc {
 			w.Header().Set("Transfer-Encoding", "chunked")
 			logStreamRequested = true
 		}
-		if job != "" && id != "" {
-			if !objReg.MatchString(job) {
-				http.Error(w, "Invalid job query", http.StatusBadRequest)
-				return
-			}
-			if !objReg.MatchString(id) {
-				http.Error(w, "Invalid ID query", http.StatusBadRequest)
+		if err := validateRequest(r); err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+		if !logStreamRequested {
+			log, err := lc.GetJobLog(job, id)
+			if err != nil {
+				http.Error(w, fmt.Sprintf("Log not found: %v", err), http.StatusNotFound)
+				logrus.WithError(err).Warning("Error returned.")
 				return
 			}
-			if !logStreamRequested {
-				log, err := lc.GetJobLog(job, id)
-				if err != nil {
-					http.Error(w, fmt.Sprintf("Log not found: %v", err), http.StatusNotFound)
-					logrus.WithError(err).Warning("Error returned.")
-					return
-				}
-				if _, err = w.Write(log); err != nil {
-					logrus.WithError(err).Warning("Error writing log.")
-				}
-			} else {
-				//run http chunking
-				options := getOptions(r.URL.Query())
-				log, err := lc.GetJobLogStream(job, id, options)
-				if err != nil {
-					http.Error(w, fmt.Sprintf("Log stream caused: %v", err), http.StatusNotFound)
-					logrus.WithError(err).Warning("Error returned.")
-					return
-				}
-				httpChunking(log, w)
+			if _, err = w.Write(log); err != nil {
+				logrus.WithError(err).Warning("Error writing log.")
 			}
 		} else {
-			http.Error(w, "Missing job and ID query", http.StatusBadRequest)
-			return
+			//run http chunking
+			options := getOptions(r.URL.Query())
+			log, err := lc.GetJobLogStream(job, id, options)
+			if err != nil {
+				http.Error(w, fmt.Sprintf("Log stream caused: %v", err), http.StatusNotFound)
+				logrus.WithError(err).Warning("Error returned.")
+				return
+			}
+			httpChunking(log, w)
 		}
 	}
 }
 
+func validateRequest(r *http.Request) error {
+	job := r.URL.Query().Get("job")
+	id := r.URL.Query().Get("id")
+
+	if job == "" {
+		return errors.New("Missing job query")
+	}
+	if id == "" {
+		return errors.New("Missing ID query")
+	}
+	if !objReg.MatchString(job) {
+		return errors.New("Invalid job query")
```

Comment:
Include the invalid content in the error message if `job` or `id` exists, but is invalid.

Reply:
sure, fixed.